specs/eventing/control-plane.md

```diff
@@ -0,0 +1,985 @@
+to the Subscription before prior to the Subscription's `Ready` condition being
+set to `true`. When a Subscription is deleted, the Channel MAY send some
+additional events to the Subscription's `spec.subscriber` after the deletion.
+
+<!--
+TODO: channel-compatible CRDs (Channelable)
+-->
+
+### Addressable Resolution
+
+Both Trigger and Subscription have optional object references (`ref` in
+`spec.subscriber`, `spec.delivery.deadLetterSink`, and `spec.reply` for
+Subscription) which are expected to conform to the Addressable partial schema
+("duck type"). An object conforms with the Addressable partial schema if it
+contains a `status.address.url` field containing a URL which may be used to
+deliver CloudEvents over HTTP. As a special case, Kubernetes `v1` `Service`
+objects are considered to have a `status.address.url` of
+`http://<service-dns-name>/`.
+
+If both the `ref` field and the `uri` fields are set on a Destination, then the
+Destination's address is the `uri` interpreted relative to the resolved URI of
+the `ref` field. This may be used, for example, to refer to a specific URL off a
+referenced domain name, like so:
+
+```yaml
+subscriber:
+  ref:
+    apiVersion: v1
+    kind: Service
+    name: test
+  uri: "/update
+```
+
+This Destination would resolve to
+`http://test.<current-namespace-dns-prefix>/update`, as `/update` would be
+interpreted relative to the `test` Service's DNS name.
+
+If one of these object references points to an object which does not currently
+satisfy this partial schema (either because the `status.address.url` field is
+empty, or because the object does not have that field), then the Trigger or
+Subscription MUST indicate an error by setting the `Ready` condition to `false`,
+and SHOULD include an indication of the error in a condition reason or type.
+
+Both Broker and Channel MUST conform to the Addressable partial schema.
+
+## Event Routing
+
+Note that the event routing description below does not cover the actual
+mechanics of sending an event from one component to another; see
+[the data plane](./data-plane.md) contracts for details of the event transfer
+mechanisms.
+
+### Content Based Routing
+
+A Broker MUST publish a URL at `status.address.url` when it is able to receive
+events. This URL MUST accept CloudEvents in both the
+[Binary Content Mode](https://github.com/cloudevents/spec/blob/v1.0.1/http-protocol-binding.md#31-binary-content-mode)
+and
+[Structured Content Mode](https://github.com/cloudevents/spec/blob/v1.0.1/http-protocol-binding.md#32-structured-content-mode)
+HTTP formats. Before sending an HTTP response, the Broker MUST durably enqueue
+the event (where durability means that the Broker can retry event delivery
+beyond the duration of receiving the event).
+
+For each event received by the Broker, the Broker MUST evaluate each associated
+Trigger **once** (where "associated" means a Trigger with a `spec.broker` which
```

Comment:
would it be wrong to say "exactly once" instead of just "once" ? I think we need to make it clear that during retries or system crashes if the Trigger is modified then the new filter should not be used. That is correct, isn't it?